chia/util/misc.py

```diff
@@ -280,3 +282,95 @@ def setup_async_signal_handler(self, handler: AsyncHandler) -> None:
         self.setup_sync_signal_handler(
             handler=functools.partial(self.threadsafe_sync_signal_handler_for_async, handler=handler)
         )
+
+
+@dataclass
+class SplitManager(Generic[T]):
```

Comment:
what is "SplitManager" used for? the name doesn't make that immediately obvious to me

Reply:
It is used in some tests.  I think they should probably be refactored so we can delete this.  Maybe I should have done that instead of this.  I dunno.

Reply:
...  it is used to allow separation of the entry and exit of a context manager in ways that they aren't meant to be used.  This is related to what we had before with separate start and stop methods.  Note the warning `# NOTE: only for transitional testing use, please avoid usage`.  Basically, I'm not fixing all the testing patterns that depended on the 'unsafe' separation of start and stop.  They can be addressed in another PR.